File: Adhiti_Shaili/Day 12/Single Number.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int num=nums[0];
        for(int i=1;i<nums.size();i++)
        {
            num=num^nums[i];
        }
        return num;
        
    }
};

File: sanket-dalvi/Day10/singleNumber.py
from functools import reduce
from operator import xor
from typing import List
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        if len(nums)==1: return nums[0]
        return reduce(xor, nums)

# Test
print(Solution().singleNumber([4,1,2,1,2]))